runtime/protocol/http-client-engines/http-client-engine-crt/common/src/aws/smithy/kotlin/runtime/http/engine/crt/RequestUtil.kt

```diff
@@ -109,3 +113,24 @@ internal suspend fun HttpStream.sendChunkedBody(body: HttpBody) {
         else -> error("sendChunkedBody should not be called for non-chunked body types")
     }
 }
+
+internal inline fun <T> mapCrtException(block: () -> T): T =
+    try {
+        block()
+    } catch (ex: CrtRuntimeException) {
+        throw HttpException(ex.message ?: ex.errorDescription, ex, errorCode = mapCrtErrorCode(ex.errorCode))
+    }
```

Comment:
**Question**: We have no signal from CRT exceptions to pass as the `retryable` argument?

Reply:
Well connect and negotiation timeouts are automatically considered retryable via `HttpException`. Are there any that stand out to you that should be default retryable?

Reply:
> I suppose it's weird to me that individual engine translation layers would be making the decision about retryability

The more localized an exception is the more information we have about whether it's retryable. As an example, what if we get a socket write timeout trying to send the request? In general a write timeout isn't retryable because the retry layer doesn't know what stage we were at in the request/response but the HTTP layer does. For the most part we aren't making decisions to retry or not in individual engines, they are mostly mapping the `HttpErrorCode` enum which contains the "generic" retryable errors logic. I think it's useful for engines (and other parts of the codebase) to be able to further influence retryability though. This removes the need to update the retry policy to explicitly cover those cases resulting in tight coupling and dependency issues.